poulpe-service/src/main/java/org/jtalks/poulpe/service/transactional/TransactionalUserService.java

```diff
@@ -247,49 +248,33 @@ public void updateUsersAtGroup(List<User> users, Group group) {
      * {@inheritDoc}
      */
     @Override
-    public PoulpeUser authenticate(String username, String password)
-            throws NotFoundException {
-        PoulpeUser user = getPoulpeUser(username);
-        if (user.getPassword().equals(password)) {
-            return user;
-        } else {
-            throw new NotFoundException();
+    public PoulpeUser authenticate(String username, String password) throws NotFoundException {
+        if (StringUtils.isAnyBlank(username, password)) throw new NotFoundException();
+        List<PoulpeUser> users = requireNotNullNorEmpty(userDao.findUsersByUsernameAndPasswordHash(username, password));
+        switch (users.size()){
+            case 1:
+                return users.get(0);
+            default:
+                return searchForUserCaseSensitive(users, username);
         }
     }
 
+    private PoulpeUser searchForUserCaseSensitive(List<PoulpeUser> users, String username) throws NotFoundException {
+        for (PoulpeUser user : users) {
+            if (user.getUsername().equals(username)) return user;
+        }
+        throw new NotFoundException();
```

Comment:
Shouldn't you pass username and password to the exception so that we see something in logs?